### DISCUSSION TITLE
build_win.bat 报错

### DISCUSSION THREAD

**Discussion by songyu1980** [deepspeedai/DeepSpeed#3239] [Q&A] [answered]
D:\DeepSpeed>build_win.bat
Administrative permissions required. Detecting permissions...
Success: Administrative permissions confirmed.
DS_BUILD_OPS=1
test.c
LINK : fatal error LNK1181: 无法打开输入文件“aio.lib”
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
Traceback (most recent call last):
  File "setup.py", line 167, in <module>
    ext_modules.append(builder.builder())
  File "D:\DeepSpeed\op_builder\builder.py", line 582, in builder
    self.build_for_cpu = not assert_no_cuda_mismatch(self.name)
  File "D:\DeepSpeed\op_builder\builder.py", line 93, in assert_no_cuda_mismatch
    raise Exception(f">- DeepSpeed Op Builder: Installed CUDA version {sys_cuda_version} does not match the "
Exception: >- DeepSpeed Op Builder: Installed CUDA version 12.1 does not match the version torch was compiled with 11.7, unable to compile cuda/cpp extensions without a matching cuda version.
D:\DeepSpeed>

**Comment by jxc-only** [comment]
你好！我也遇到了同样的问题，你的问题解决了吗？能否分享一下解决方案

**Comment by loadams** [comment, marked as answer]
The exception you are hitting is that you've installed cuda 12.1 and torch+cuda 11.7.  You can fix this by installing a torch version that is compiled with cuda 12.1 libs.

**Reply by under-shadow** [reply]
如果我是想下载CPU版本，也出现了LINK：致命错误 LNK1181：无法打开输入文件“aio.lib”，有没有办法解决

**Reply by loadams** [reply]
The async_io library isn't supported on Windows, so that error is expected but will not cause issues.  There should be another error, can you please share the full output log?
